Source/WebCore/Modules/url-pattern/URLPattern.cpp

```diff
@@ -242,6 +243,27 @@ ExceptionOr<Ref<URLPattern>> URLPattern::create(std::optional<URLPatternInput>&&
 
     auto processedInit = maybeProcessedInit.releaseReturnValue();
 
+    if (!processedInit.protocol)
+        processedInit.protocol = "*"_s;
+    if (!processedInit.username)
+        processedInit.username = "*"_s;
+    if (!processedInit.password)
+        processedInit.password = "*"_s;
+    if (!processedInit.hostname)
+        processedInit.hostname= "*"_s;
+    if (!processedInit.pathname)
+        processedInit.pathname = "*"_s;
+    if (!processedInit.search)
+        processedInit.search = "*"_s;
+    if (!processedInit.hash)
+        processedInit.hash = "*"_s;
+    if (!processedInit.port) {
+        if (isSpecialScheme(processedInit.protocol) && isDefaultPortForProtocol(*parseInteger<uint16_t>(processedInit.port), processedInit.protocol))
```

Comment:
If `processInit.port` is null, then why do we still need to check if it's default port?
According to spec, the `isSpecialScheme` check should happen outside of the ` if (!processedInit.port) { ... }`

Reply:
I thought it was easier / more readable to combine. 

But you're right maybe it's probably confusing if I deviate from the spec. I will change it back then!